app/import.go

```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+const maxScanTokenSize = 16 * 1024 * 1024 // Need to set a higher limit than default because some customers cross the limit. See MM-22314
```

Comment:
Are we confident that allowing 67MB tokens into memory won't crash certain customer systems? That sounds quite big. Can we run some tests on a system following our official system specs with a realistic import file that is on the high end?

Reply:
Isn't it 16MB ? Or maybe I am missing something.

Reply:
No, you are absolutely right - I just did completely wrong math. :-) Just to be sure, does the scanner only keep a single token in memory? In a worst case scenario, is there any risk our memory usage would explode if there would be plenty of tokens of 16mb size? I just want to avoid that we avoid risking instability for lower spec'ed environments.

Reply:
Correct. One token at a time. It starts with the initial token size, and doubles up if it encounters bigger tokens. Till a limit of 16MB. If the token size exceeds 16MB, it will throw "bufio.Scanner: token too long". That's as per my understanding.

So at no point in time, can the memory be more than 16MB.

Reply:
Thanks a lot for the clarification Agniva, sounds good to me!